File: result/repair/starchat/python2java/python_repair/SUM_SQUARES_BINOMIAL_COEFFICIENTS.py
def f_gold(n):
        C = [[0 for _ in range(n + 1)] for _ in range(n + 1)]
        for i in range(n + 1):
            j=0 
            while j<min(i,n)+1 :
                if j == 0 or j == i:
                    C[i][j] = 1
                else:
                    C[i][j] = C[i - 1][j - 1] + C[i - 1][j]
                j += 1
        sum = 0
        for i in range(n + 1):
            sum += C[n][i] * C[n][i]
        return sum

File: result/translate/chatgpt/java2cpp/c++/SUM_SQUARES_BINOMIAL_COEFFICIENTS.cpp
int f_gold(int n) {
    int** C = new int*[n + 1];
    for (int i = 0; i <= n; i++) {
        C[i] = new int[n + 1];
    }
    int i, j;
    for (i = 0; i <= n; i++) {
        for (j = 0; j <= std::min(i, n); j++) {
            if (j == 0 || j == i) C[i][j] = 1;
            else C[i][j] = C[i - 1][j - 1] + C[i - 1][j];
        }
    }
    int sum = 0;
    for (i = 0; i <= n; i++) sum += (C[n][i] * C[n][i]);
    return sum;
}